### PR TITLE
Migrate readthedocs to version 2

### PR DESCRIPTION
**Pull request by pablocasares**
## Description

Migrate readthedocs to version 2.

## Motivation and Context

Readthedocs docs generation stopped working because old setup was deprecated.

### CODE REVIEW

**Comment by honnix** [comment]
Is this verified for PR build? Is it possible to show that in the PR checks?

File: .readthedocs.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+build:
+  os: ubuntu-22.04
+  tools:
+    python: "3.9"
+
+sphinx:
+  configuration: doc/conf.py
+
+formats:
+  - pdf
```
Comment (honnix) [inline .readthedocs.yaml:12]: Do we need these format?

Comment (pablocasares) [reply .readthedocs.yaml:12]: they were enabled in the old setup. We can disable them if not needed.

Comment (honnix) [reply .readthedocs.yaml:12]: I see. Let's keep them then.

**Review by honnix — DISMISSED** [review]

**Comment by pablocasares** [comment]
> Is this verified for PR build? Is it possible to show that in the PR checks?

I updated the setup and it should now show the step in PR builds.

**Review by honnix — APPROVED** [review]

**Pull request merged**